src/rule_definition_tools.jl

```diff
@@ -392,7 +396,70 @@ function _nondiff_rrule_expr(__source__, primal_sig_parts, primal_invoke)
 end
 
 
-###########
+############################################################################################
+# @opt_out
+
+"""
+    @opt_out frule([config], _, f, args...)
+    @opt_out rrule([config], f, args...)
+
+This allows you to opt-out of a `frule` or `rrule` by providing a more specific method,
+that says to use the AD system, to solver it.
+
+For example, consider some function `foo(x::AbtractArray)`.
+In general, you know a efficicent and generic way to implement it's `rrule`.
+You do so, (likely making use of [`ProjectTo`](@ref)).
+But it actually turns out that for some `FancyArray` type it is better to let the AD do it's
+thing.
+
+Then you would write something like:
+```julia
+function rrule(::typeof(foo), x::AbstractArray)
+    foo_pullback(ȳ) = ...
+    return foo(x), foo_pullback
+end
+
+@opt_out rrule(::typeof(foo), ::FancyArray)
+```
+
+This will generate a [`rrule`](@ref) that returns `nothing`,
+and will also add a similar entry to [`ChainRulesCore.no_rrule`](@ref).
```

Comment:
It is part of the public API.
It probably should not be described as a implementation detail

Reply:
Oh, right, it's part of the public API because we want the AD systems to use it - that makes sense. Ignore my other comment which assumes the same thing